src/Entities/Aero/AeroBlockSlingshot.cs

```diff
@@ -0,0 +1,417 @@
+            data.Bool("allowAdjustments", DefaultAllowAdjustments),
+            data.Attr("startColor", DefaultStartColor),
+            data.Attr("endColor", DefaultEndColor))
+    {
+    }
+
+    public AeroBlockSlingshot(Vector2[] positions, int width, int height,
+        float launchTime = DefaultLaunchTime,
+        float cooldownTime = DefaultCooldownTime,
+        float setTime = DefaultSetTime,
+        float delayTime = DefaultDelayTime,
+        float pushSpeed = DefaultPushSpeed,
+        Pushable.MoveActionType pushActions = DefaultPushActions,
+        bool allowAdjustments = DefaultAllowAdjustments,
+        string startColor = DefaultStartColor,
+        string endColor = DefaultEndColor)
+        : base(positions[0], width, height)
+    {
+        LaunchTime = launchTime;
+        CooldownTime = cooldownTime;
+        SetTime = setTime;
+        DelayTime = delayTime;
+        PushSpeed = pushSpeed;
+        AllowAdjustments = allowAdjustments;
+        StartColor = Calc.HexToColor(startColor);
+        EndColor = Calc.HexToColor(endColor);
+
+        startPosition = Position;
+
+        for (int i = 0; i < positions.Length; i++)
+            positions[i].Y = startPosition.Y;
+
+        this.positions = positions;
+
+        sortedPositions = positions.OrderBy(p => p.X).ToArray();
+        leftPosition = sortedPositions.First();
+        rightPosition = sortedPositions.Last();
+
+        Add(pushable = new Pushable
+        {
+            OnPush = OnPush,
+            PushCheck = PushCheck,
+            MaxPushSpeed = PushSpeed,
+            Active = false,
+            MoveActions = pushActions,
+        });
+
+        Add(sfx = new SoundSource()
+        {
+            Position = new Vector2(width, height) / 2.0f,
+        });
+
+        stateMachine = new StateMachine();
+        stateMachine.SetCallbacks((int) SlingshotStates.Idle, IdleUpdate, begin: IdleBegin, end: IdleEnd);
+        stateMachine.SetCallbacks((int) SlingshotStates.Windup, WindupUpdate, begin: WindupBegin, end: WindupEnd);
+        stateMachine.SetCallbacks((int) SlingshotStates.Locked, LockedUpdate, begin: LockedBegin, end: LockedEnd);
+        stateMachine.SetCallbacks((int) SlingshotStates.Launch, LaunchUpdate, begin: LaunchBegin, end: LaunchEnd);
+        stateMachine.SetCallbacks((int) SlingshotStates.Cooldown, CooldownUpdate, begin: CooldownBegin, end: CooldownEnd);
+        stateMachine.State = (int) SlingshotStates.Idle;
+        Add(stateMachine);
+
+        progressScreen = new((int) Width, (int) Height)
+        {
+            Color = Color.Tomato,
+        };
+    }
+
+    public override void Update()
+    {
+        base.Update();
+        updateTimer -= Engine.DeltaTime;
+    }
+
+    private void IdleBegin()
+    {
+        pushable.Active = true;
+
+        if (sfx.Playing)
+            sfx.Stop();
+    }
+
+    private int IdleUpdate()
+    {
+        return (int) SlingshotStates.Idle;
+    }
+
+    private void IdleEnd()
+    {
+    }
```

Comment:
AFAIK It is possible to remove those and pass `null` to `StateMachine.SetCallbacks` where they are passed. But honestly I don't mind if they're left here for clarity

Reply:
Yeah I put all the stubs in initially in case I needed them, but I forgot to go through and remove them afterward.